pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -234,6 +234,17 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
 			return reconcile.Result{RequeueAfter: timeout}, nil
 		}
 		removeFinalizerFromCRDInstance(ctx, instance)
+		err = updateCnsFileAccessConfig(ctx, r.client, instance)
+		if err != nil {
+			msg := fmt.Sprintf("failed to update CnsFileAccessConfig instance: %q on namespace: %q. Error: %+v",
+				instance.Name, instance.Namespace, err)
+			recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		// Cleanup instance entry from backOffDuration map
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
```

Comment:
I see this code block is repeated several times in this method. This can be simplified if an internal function is written for the whole logic that returns an error in case of error conditions and returns success otherwise. A common code can act on the return variables. I think you can also handle the backoff logic in a nicer way.